packages/datadog-plugin-jest/src/index.js

```diff
@@ -0,0 +1,167 @@
+const RESOURCE_NAME = 'resource.name'
+
+function getCIMetadata () {
+  const { env } = process
+  if (env.GITHUB_ACTIONS) {
+    const { GITHUB_REF, GITHUB_SHA, GITHUB_REPOSITORY, GITHUB_RUN_ID, GITHUB_RUN_NUMBER, GITHUB_WORKSPACE } = env
+
+    const pipelineURL = `https://github.com/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}`
+
+    return {
+      [CI_PIPELINE_URL]: pipelineURL,
+      [CI_PIPELINE_ID]: GITHUB_RUN_ID,
+      [CI_PROVIDER_NAME]: 'github',
+      [CI_PIPELINE_NUMBER]: GITHUB_RUN_NUMBER,
+      [CI_WORKSPACE_PATH]: GITHUB_WORKSPACE,
+      [GIT_BRANCH]: GITHUB_REF,
+      [GIT_COMMIT_SHA]: GITHUB_SHA
+    }
+  }
+  return {}
+}
+
+const sanitizedRun = async cmd => {
+  try {
+    return (await promiseExec(cmd)).stdout.replace(/(\r\n|\n|\r)/gm, '')
+  } catch (e) {
+    return ''
+  }
+}
+
+async function getGitInformation () {
+  return {
+    repository: await sanitizedRun('git ls-remote --get-url'),
+    branch: await sanitizedRun('git branch --show-current'),
+    commit: await sanitizedRun('git rev-parse HEAD')
+  }
+}
+
+function finishStartedSpans () {
+  global.tracer
+    .scope()
+    .active()
+    .context()._trace.started.forEach((span) => {
+      span.finish()
+    })
+}
+
+function setTestStatus (status) {
+  global.tracer.scope().active().setTag(TEST_STATUS, status)
+}
+
+function getEnvironment (BaseEnvironment) {
+  return class DatadogJestEnvironment extends BaseEnvironment {
+    constructor (config, context) {
+      super(config, context)
+      this.testSuite = context.testPath.replace(`${config.rootDir}/`, '')
+      this.rootDir = config.rootDir
+    }
+    async setup () {
+      if (!global.tracer) {
+        const ciMetadata = getCIMetadata()
+        const { repository, branch, commit } = await getGitInformation()
+        global.tracer = require('../../dd-trace').init({
+          sampleRate: 1,
+          flushInterval: 1,
+          startupLogs: false,
+          ingestion: {
+            sampleRate: 1
+          },
+          tags: {
+            ...ciMetadata,
+            [GIT_COMMIT_SHA]: commit,
+            [GIT_BRANCH]: branch,
+            [GIT_REPOSITORY_URL]: repository,
+            [BUILD_SOURCE_ROOT]: this.rootDir,
+            [TEST_FRAMEWORK]: 'jest'
+          }
+        })
+      }
+
+      return super.setup()
+    }
+    async teardown () {
+      await new Promise((resolve) => {
+        global.tracer._tracer._exporter._writer.flush(resolve)
+      })
+      return super.teardown()
+    }
+
+    async handleTestEvent (event) {
+      if (event.name === 'test_skip' || event.name === 'test_todo') {
+        global.tracer.startSpan(
+          'jest.test',
+          {
+            tags: {
+              // Since we are using `startSpan` we can't use `type` and `resource` options
+              // so we have to manually set the tags.
+              [SPAN_TYPE]: 'test',
+              [RESOURCE_NAME]: `${event.test.parent.name}.${event.test.name}`,
```

Comment:
as far as I can see `SpanOptions` does not allow for `type` or `resource` inputs so you'd have to manually set `span.type` and `resource.name`, right?

Reply:
Correct.